pkg/codegen/customresourcegenerator.go

```diff
@@ -46,45 +54,189 @@ type CustomResourceGenerator struct {
 	ResourceTokens []string
 }
 
-func NewCustomResourceGenerator(crd unstructured.Unstructured) (CustomResourceGenerator, error) {
-	apiVersion := crd.GetAPIVersion()
-	schemas := map[string]map[string]any{}
-
-	validation, foundValidation, _ := unstructured.NestedMap(crd.Object, "spec", "validation", "openAPIV3Schema")
-	if foundValidation { // If present, use the top-level schema to validate all versions
-		versionName, foundVersionName, _ := unstructured.NestedString(crd.Object, "spec", "version")
-		if foundVersionName {
-			schemas[versionName] = validation
-		} else if versionInfos, foundVersionInfos, _ := unstruct.NestedMapSlice(crd.Object, "spec", "versions"); foundVersionInfos {
-			for _, versionInfo := range versionInfos {
-				versionName, _, _ := unstructured.NestedString(versionInfo, "name")
-				schemas[versionName] = validation
-			}
+// flattenOpenAPI recursively finds all nested objects in the OpenAPI spec and flattens them into a single object as definitions.
+func flattenOpenAPI(sw *spec.Swagger) error {
```

Comment:
This is great -- the surface area for potential issues is signifcantly smaller. If we have bugs they'll almost certainly either live in this flattening logic or the upstream p-p or p-k codegen.